.golangci.yml

```diff
@@ -94,45 +89,59 @@ linters:
     # we don't want to change code to Go 1.22+ yet
     - intrange
     - copyloopvar
-    - tenv
 
-linters-settings:
-  goconst:
-    min-len: 2
-    min-occurrences: 2
-  gocritic:
-    enabled-tags:
-      - diagnostic
-      - experimental
-      - opinionated
-      - performance
-      - style
-    disabled-checks:
-      - dupImport # https://github.com/go-critic/go-critic/issues/845
-      - ifElseChain
-      - whyNoLint
-      - sloppyReassign
-      - uncheckedInlineErr # Experimental rule with high false positive rate.
-  gocyclo:
-    min-complexity: 15
-  govet:
-    enable-all: true
-    disable:
-      - fieldalignment
-    settings:
-      printf: # analyzer name, run `go tool vet help` to see all analyzers
-        funcs: # run `go tool vet help printf` to see available settings for `printf` analyzer
-          - debug,debugf,debugln
-          - error,errorf,errorln
-          - fatal,fatalf,fataln
-          - info,infof,infoln
-          - log,logf,logln
-          - warn,warnf,warnln
-          - print,printf,println,sprint,sprintf,sprintln,fprint,fprintf,fprintln
-  lll:
-    line-length: 100
-    tab-width: 4
-  misspell:
-    locale: US
-    ignore-words:
-      - rela
+  settings:
+    goconst:
+      min-len: 2
+      min-occurrences: 2
+    gocritic:
+      enabled-tags:
+        - diagnostic
+        - experimental
+        - opinionated
+        - performance
+        - style
+      disabled-checks:
+        - dupImport # https://github.com/go-critic/go-critic/issues/845
+        - ifElseChain
+        - whyNoLint
+        - sloppyReassign
+        - uncheckedInlineErr # experimental rule with high false positive rate.
+        - importShadow # shadow of imported package
+    gocyclo:
+      min-complexity: 15
+    gosec:
+      excludes:
+        - G103 # unsafe calls should be audited
+        - G104 # unhandled errors
```

Comment:
Suppressing to point out unhandled errors seems to be a downgrade.
Unhanded errors are often the cause for issues and we should not ignore them.
```suggestion
```
What is the reasoning to ingore unhanded errors?

Reply:
We can silence these errors as well with (for example)
```
_ = f.Close()
```
But it seems to add noise to the source code.

Though there are reasons why we want to enable this (and other) linter settings in the future... but that should be separate issues/PRs.

Reply:
> But it seems to add noise to the source code.

I don't agree with that. With `nolint:gosec` the author of a code explicitly calls out, that they accept the risk of ignoring a certain case. And it is easier for reviewer to spot such cases.
Disabling this check puts a higher burden on the reviewers.

Reply:
Enabled G104